pkg/controller/pravegacluster/upgrade.go

```diff
@@ -404,6 +405,178 @@ func (r *ReconcilePravegaCluster) syncSegmentStoreVersion(p *pravegav1alpha1.Pra
 	return false, nil
 }
 
+//this function is to check are we doing a rollback in case of a upgrade failure while upgrading from a version below 07 to a version above 07
+func (r *ReconcilePravegaCluster) IsClusterRollbackingFrom07(p *pravegav1alpha1.PravegaCluster) bool {
+	if !r.isRollbackTriggered(p) {
+		return false
+	}
+	//checking if sts above07 exsists
+	stsAbove07 := &appsv1.StatefulSet{}
+	name := util.StatefulSetNameForSegmentstoreAbove07(p.Name)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, stsAbove07)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		log.Printf("failed to get PravegaCluster: %v", err)
+		return false
+	}
+
+	//checking if sts below07 exsists
+	stsBelow07 := &appsv1.StatefulSet{}
```

Comment:
How are we dealing with situations where one STS is deleted and other is present?

Reply:
that's not possible as we are not deleting the oldsts  till the time new sts hasn't come up completely meaning all it's ready replicas are equal to segmentstore replicas